crates/telio-model/src/config.rs

```diff
@@ -194,6 +194,18 @@ pub struct Config {
     pub dns: Option<DnsConfig>,
 }
 
+impl Config {
+    /// Check if the config is empty
+    pub fn is_empty(&self) -> bool {
+        self.this.hostname.is_empty()
+            && self.this.identifier.is_empty()
+            && self.this.ip_addresses.is_none()
```

Comment:
Any reason to omit: `self.this.public_key`?

Reply:
Public key is non optional and it itself is wrapper over `[u8; KEY_SIZE]` so it never is empty (unless you would count something like `[0; KEY_SIZE]` as empty).

Reply:
I think it was mainly a decision of convenience, influenced by exactly what Tomasz is saying. It will never be empty so the next best thing is to check that it's all 0, but if everything else in the config is empty, we can already say that the config is empty and skip that check. It's not the cleanest solution but it was the most convenient, and one made to keep the interface more similar. Though it's probably better, now that we don't deserialize strings into configs to not have the clients send None instead of an empty config if that's what they mean. After all, the clients should now better in which scenarios the config is empty by intention